packages/browser-wallet/src/background/recovery.ts

```diff
@@ -4,36 +4,43 @@ import {
     createIdentityRecoveryRequest,
     CredentialInputV1,
     CredentialRegistrationId,
-    CryptographicParameters,
     HttpProvider,
     IdentityRecoveryRequestInput,
     JsonRpcClient,
-    Network,
 } from '@concordium/web-sdk';
-import { ExtensionMessageHandler, InternalMessageType } from '@concordium/browser-wallet-message-hub';
+import { InternalMessageType } from '@concordium/browser-wallet-message-hub';
 import { BackgroundResponseStatus, RecoveryBackgroundResponse } from '@shared/utils/types';
-import { Identity, CreationStatus, IdentityProvider, WalletCredential } from '@shared/storage/types';
-import { sessionIsRecovering, storedCredentials, storedCurrentNetwork, storedIdentities } from '@shared/storage/access';
+import { Identity, CreationStatus, IdentityProvider, WalletCredential, RecoveryStatus } from '@shared/storage/types';
+import {
+    sessionIsRecovering,
+    sessionRecoveryStatus,
+    storedCredentials,
+    storedCurrentNetwork,
+    storedIdentities,
+} from '@shared/storage/access';
 import { identityMatch, isIdentityOfCredential } from '@shared/utils/identity-helpers';
 import { getNextUnused } from '@shared/utils/number-helpers';
-import { addCredential, addIdentity } from './update';
+import { partition } from 'wallet-common-helpers';
+import { addCredential, addIdentity, updateCredentials } from './update';
 import bgMessageHandler from './message-handler';
 import { openWindow } from './window-management';
 
 // How many empty identityIndices are allowed before stopping
 const maxEmpty = 20;
+const RECOVERY_ALARM_NAME = 'recoveryAlarm';
+const RECOVERY_LOCK = 'recoveryLock';
```

Comment:
The current lock naming convention is:
```suggestion
const RECOVERY_LOCK = 'concordium_recovery_lock';
```
Perhaps all the lock constants should be collected to a single file (at some point in the future).